test/lib/jdk/test/lib/util/JarUtils.java

```diff
@@ -86,7 +87,11 @@ public static void createJarFile(Path jarfile, Manifest man, Path dir, Path... f
             for (Path entry : entries) {
                 String name = toJarEntryName(entry);
                 jos.putNextEntry(new JarEntry(name));
-                Files.copy(dir.resolve(entry), jos);
+                if (Files.exists(dir.resolve(entry))) {
```

Comment:
This is test infrastructure that we use in several areas and changing it to allow file paths to files that don't exist be problematic. Is there any reason why the jarsigner can't create an empty or dummy file to put into the JAR file?

Reply:
Sorry, I'll revert the change and create files myself. I just thought any existing call to this method should have the file already created there, but it could be a problem if the creation is not trivial and might fail.